src/Broadway/CommandHandling/ClosureCommandHandler.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of the broadway/broadway package.
+ *
+ * (c) Qandidate.com <opensource@qandidate.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Broadway\CommandHandling;
+
+use Broadway\CommandHandling\Exception\ClosureParameterNotAnObjectException;
+
+/**
+ * Using this class command can be registered with closure
+ */
+abstract class ClosureCommandHandler implements CommandHandler
+{
+    private $handlers;
+
+    public function __construct()
+    {
+        $this->configure();
+    }
+
+    abstract protected function configure();
+
+    /**
+     * @param \Closure $handler
+     */
+    public function add(\Closure $handler)
+    {
+        $reflection = new \ReflectionFunction($handler);
+        $reflectionParams = $reflection->getParameters();
+
+        if(!isset($reflectionParams[0]) || !$reflectionParams[0]->getClass()) {
```

Comment:
$reflectionParams variable it's needed in the line below. I don't like to create an assert method that return a variable.